lib/form1010_ezr/service.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+require 'hca/enrollment_system'
+require 'hca/configuration'
+require 'va_1010_forms/service_utils'
+
+module Form1010Ezr
+  class Service < Common::Client::Base
+    include VA1010Forms::ServiceUtils
+
+    # Due to using the same endpoint as the 10-10EZ (HealthCareApplication), we
+    # can utilize the same client configuration
+    configuration HCA::Configuration
+
+    FORM_ID = '10-10EZR'
+
+    # @param [Object] user
+    def initialize(user)
+      super()
+      @user = user
+    end
+
+    # @param [HashWithIndifferentAccess] parsed_form JSON form data
+    def submit_form(parsed_form)
+      begin
+        validate_form(parsed_form)
+
+        formatted = HCA::EnrollmentSystem.veteran_to_save_submit_form(parsed_form, @user)
+        content = Gyoku.xml(formatted)
+        submission = soap.build_request(:save_submit_form, message: content)
+
+        response = perform(:post, '', submission.body)
+      rescue => e
+        Rails.logger.error "10-10EZR form submission failed: #{e.message}"
+        raise e
+      end
+
+      root = response.body.locate('S:Envelope/S:Body/submitFormResponse').first
+
+      {
+        success: true,
+        formSubmissionId: root.locate('formSubmissionId').first.text.to_i,
+        timestamp: root.locate('timeStamp').first&.text || Time.now.getlocal.to_s
+      }
+    end
+
+    private
+
+    # Compare the 'parsed_form' to the form schema in vets-json-schema
+    def validate_form(parsed_form)
```

Comment:
validation should be done the same way it's done here
https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/controllers/v0/preneeds/burial_forms_controller.rb#L65
don't raise standarderror for validation errors

Reply:
Okay, I cleaned it up and added the schema validation error